### PR TITLE
Allow skipping notability checks using label.

### PR DESCRIPTION
**Pull request by reitermarkus**
Needed for cases like https://github.com/Homebrew/homebrew-cask/pull/130308.

### CODE REVIEW

**Comment by SMillerDev** [comment]
Do we really need a label though? As a maintainer we can just choose to merge regardless.

**Comment by reitermarkus** [comment]
I am of the opinion that no PRs should be failing CI before merging, so applying a label instead of merging a failing PR is preferable.

**Comment by SMillerDev** [comment]
Might be better to tweak the warning level then. Since a label to disable a test isn't really fixing the issue. It's just hiding it.

**Comment by Bo98** [comment]
This would be beneficial to have so that we can re-enable requiring green CI here. I know things have been accidentally merged because they appear green during a GitHub Actions outage.

**Comment by reitermarkus** [comment]
> Might be better to tweak the warning level then. Since a label to disable a test isn't really fixing the issue. 

I'm not sure what changing the warning level would do in this case?

The specific problem in the case of https://github.com/Homebrew/homebrew-cask/pull/130308 is that the repository used for hosting the downloads differs from the code for the source code of the application, which has vastly more stars and forks.

> It's just hiding it.

The label should of course only be used in those cases which absolutely require it. And in doing so, it's doing the opposite of hiding it, it's showing the manual review of a maintainer as a visible label.

**Comment by reitermarkus** [comment]
> Being able to override audit failures was one of the major blockers that required that enforcement to be disabled

Yeah, I don't agree with the decision to disable required CI checks. In my opinion, if CI is failing, it either needs to be fixed or an exception added.

**Comment by bevanjkay** [comment]
> I am of the opinion that no PRs should be failing CI before merging, so applying a label instead of merging a failing PR is preferable.

I agree with this sentiment. It makes for a more verbose approach, as it is an express statement to say the maintainer is aware of the audit failure but intends for the PR to be merged regardless. Where just clicking squash and merge on a failing PR shows that potentially the maintainer was aware, but also may have just not noticed...

File: cmd/lib/ci_matrix.rb [hunk 1]
```diff
@@ -105,13 +105,10 @@ def self.generate(tap, labels: [])
...
-      appcast_arg = if labels.include?("ci-skip-appcast")
-        "--no-appcast"
-      else
-        "--appcast"
-      end
+      audit_args = ["--online"]
 
-      audit_args = [appcast_arg, "--online"]
+      audit_args += ["--no-appcast"] if labels.include?("ci-skip-appcast")
+      audit_args += ["--skip-repository"] if labels.include?("ci-skip-repository")
```
Comment (carlocab) [inline cmd/lib/ci_matrix.rb:111]:
```suggestion
      audit_args << "--no-appcast" if labels.include?("ci-skip-appcast")
      audit_args << "--skip-repository" if labels.include?("ci-skip-repository")
```
Optional style nit.

**Review by carlocab — APPROVED** [review]

**Comment by Bo98** [comment]
> 2\. allows you to merge PRs without ever even looking at them (when, previously, the repo PR requirements required you to at least look at the "Files" tab to approve)

Approval is still a requirement for merging, unless you are counting `brew pr-publish`.

**Comment by carlocab** [comment]
> Approval is still a requirement for merging, unless you are counting `brew pr-publish`.

Has that setting been applied to other `homebrew-cask-*` repos? The "Squash and merge" button is green for me here, and there are no reviews: https://github.com/Homebrew/homebrew-cask-fonts/pull/6223

**Comment by Bo98** [comment]
It seems not. Not sure why. I've applied it to all cask repos now - let me know if something breaks.

**Comment by MikeMcQuaid** [comment]
I like this idea, nice work @reitermarkus. Would be good to adopt for formulae, too.

**Comment by MikeMcQuaid** [comment]
Let's close this out and reopen it when it's ready for merge.

**Pull request closed without merging**